### ISSUE TITLE
http4s pathVar marked as error in IntelliJ

### ISSUE THREAD

**Issue by randm-ch** [mblink/routing#3]
![Screenshot from 2021-01-04 13-09-42](https://user-images.githubusercontent.com/2759010/103533895-37989080-4e8e-11eb-9893-a16876275f1a.png)
![Screenshot from 2021-01-04 13-39-00](https://user-images.githubusercontent.com/2759010/103537535-e0e28500-4e94-11eb-9770-43c9fc84cc2b.png)

IntelliJ infers route/query vars on http4s routes and implicits on reverse routes wrongly, causing errors. Not sure if there's something you can do on your end, if not I opened https://youtrack.jetbrains.com/issue/SCL-18591.

Thanks for your work. Cheers!

**Comment by mrdziuban** [comment]
Thanks for the report @randm-ch and apologies for the delay! I believe these issues relate to the way parameters were represented as nested tuples. I just merged #5 so parameters are now flattened when the route is built:

```scala
GET / pathVar[Int]("id") :? queryParam[String]("foo") // before - Route[GET.type, ((Unit, Int), String)]
GET / pathVar[Int]("id") :? queryParam[String]("foo") // after  - Route[GET.type, (Int, String)]
```

These changes have been released as v1.0.0 which also includes

- Scala.js support
- Cross-building for several different http4s versions
- Hosting on GitHub instead of Bintray (as [Bintray is being sunset](https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/)) -- note new resolver in example below

When you have a chance could you try out the update and see if it resolves your IntelliJ issues?

```scala
resolvers += "bondlink-maven-repo" at "https://raw.githubusercontent.com/mblink/maven-repo/main"
libraryDependencies ++= Seq(
  "bondlink" %% "routing-core" % "1.0.0",
  // choose one of these
  "bondlink" %% "routing-http4s_0.21" % "1.0.0",      // latest stable version
  "bondlink" %% "routing-http4s_1.0.0-M10" % "1.0.0", // latest version on cats-effect 2
  "bondlink" %% "routing-http4s_1.0.0-M19" % "1.0.0", // latest version on cats-effect 3
)
```

**Comment by randm-ch** [comment]
Awesome, thanks a million!
